setup.py

```diff
@@ -19,17 +19,24 @@ def get_description():
     with open('README.rst', 'r') as f:
         return ''.join(f.readlines()[2:])
 
-requires = [
-    'python-fedora',
-    'fedmsg',
-    'fedmsg_meta_fedora_infrastructure',
-    'dogpile.cache',
-    'six',
-]
-
-tests_require = [
-    'nose',
-]
+
+def get_requirements(filename='requirements.txt'):
+    """
+    Get the contents of a file listing the requirements.
+
+    :param requirements_file: path to a requirements file
+    :type  requirements_file: str
+
+    :returns: the list of requirements
+    :return type: list
+    """
+    with open(filename) as f:
+        return [
+            line.rstrip().split('#')[0]
+            for line in f.readlines()
+            if not line.startswith('#')
```

Comment:
It is a little, but I wanted it to behave like other projects (like https://pagure.io/pagure/blob/master/f/setup.py) so people wouldn't try something they can do elsewhere and be surprised.